sc-tools/sc-builder/tests/CMakeLists.txt

```diff
@@ -1,13 +1,21 @@
-configure_file(
-    "${CMAKE_CURRENT_SOURCE_DIR}/test_defines.hpp.in"
-    "${CMAKE_CURRENT_SOURCE_DIR}/units/test_defines.hpp"
+add_custom_target(sc-builder-tests-kb-generator ALL
+    COMMAND "${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/sc-builder" -i "${CMAKE_CURRENT_SOURCE_DIR}/repo.path" -o "${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/sc-builder-test-repo" --clear
+    DEPENDS sc-builder
 )
 
 make_tests_from_folder(${CMAKE_CURRENT_SOURCE_DIR}/units
     NAME sc-builder-tests
     DEPENDS sc-builder-lib
     INCLUDES ${SC_BUILDER_SRC}
 )
+add_dependencies(sc-builder-tests sc-builder-tests-kb-generator)
+target_compile_definitions(sc-builder-tests PRIVATE
+    SC_BUILDER_KB_BIN="${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/sc-builder-test-repo"
+    SC_BUILDER_REPO_PATH="${CMAKE_CURRENT_SOURCE_DIR}/repo.path"
+    SC_BUILDER_KB="${CMAKE_CURRENT_SOURCE_DIR}/kb"
+    SC_BUILDER_INI="${CMAKE_CURRENT_SOURCE_DIR}/sc-builder-test.ini"
+    SC_BUILDER_TEST_REPOS="${CMAKE_CURRENT_SOURCE_DIR}/repos"
+)
```

Comment:
Nope. No way. I'm not letting this code hit the main branch. It should be a pre-execution (also called pre-test) target relative to the sc-builder-tests, not a pre-build dependency. Calling our code in the build stage is a fire hazard since it can fail the build process due to tight timeouts in Ninja, not to say that this is straight up inefficient to do on every rebuild & rebuild would be required to check the test with the modified KB, and that is a rather bad developer experience.

FWIW, this logic should be placed inside the `SetUp` method of the test fixtures and not inside the build system as the build system should have no ties to the actual logic of the tests, injecting working directories and build settings are the only things build system is responsible for in regard to tests. 

By the way, `gtest_discover_tests` has a `WORKING_DIRECTORY` attribute that would make all the compile definitions obsolete since you could then use paths relative to the test folder in test code. 

This and all similar tests should be refactored with points I've made above in mind, the current way of doing this should be documented as discouraged, an example of the correct way to create a test that has to use a pre-compiled KB has to be documented as well.